core/NameRef.h

```diff
@@ -74,9 +104,11 @@ class NameRef final : private DebugOnlyCheck<NameRefDebugCheck> {
         return _id;
     }
 
-    NameData data(GlobalState &gs) const;
+    NameKind kind(const GlobalState &gs) const;
 
-    const NameData data(const GlobalState &gs) const;
+    const UniqueNameData unique(const GlobalState &gs) const;
+    const RawNameData raw(const GlobalState &gs) const;
```

Comment:
What is a RawName? Does this correspond to a `utf8` name kind? Can we pick one name and use that everywhere (in the enum class and the `*Data` class)?

Reply:
(I'm fine pushing this into a separate PR if you want)

Reply:
Yeah, I debated calling it `UTF8Name`. I can change that easily. I called it `RawName` because the old union type property was `.raw`.

Reply:
I went all-in on UTF8. Pushed.